Add Jenkinsfile for building docker image

dbnicholson opened the pull request:
This will build the image and push it to the endlessm/eos-idp repository
on Docker Hub. The image tag will be `pr<number>` for a pull request or
the git branch name. An additional tag such as `latest` can be provided
via the `TAG` job parameter.

dbnicholson committed:
Add Jenkinsfile for building docker image

This will build the image and push it to the endlessm/eos-idp repository
on Docker Hub. The image tag will be `pr<number>` for a pull request or
the git branch name. An additional tag such as `latest` can be provided
via the `TAG` job parameter.

eos-shell-build commented:
I don't recognize the submitter as a member of @endlessm. Can somebody in @endlessm verify the patch? See https://phabricator.endlessm.com/w/software/build/ci-integration/#pull-requests for more details.

dbnicholson commented:
ok to test

dbnicholson commented:
test this please

eos-shell-build commented:
Build finished.

dbnicholson commented:
test this please

dbnicholson commented:
test this please

dbnicholson commented:
test this please

dbnicholson commented:
test this please

dbnicholson committed:
Add settings file for testing

Currently this just ensures that SECRET_KEY is set since the default
settings are paranoid and expect this from an external source.

dbnicholson commented:
This is working now as can be seen in the checks that I have wired up to a temporary job. The other parts are https://github.com/endlessm/eos-administration/pull/992 and https://github.com/endlessm/eos-build/pull/1196.

As noted in the eos-administration job, the reason to prefer this over the Docker Hub automated builds (which I'm currently using for eos-idp) is that it allows more of a CI/CD pipeline. Jenkins is where we'd be doing the deployment, and doing the image build there as well allows it to gate the deployment. If the build fails on Docker Hub we have no way to know and might be deploying a stale image.

adarnimrod reviewed (approved):
LGTM.

dbnicholson commented:
retest this please

dbnicholson commented:
This worked in [jenkins](https://ci.endlessm-sf.com/view/infra-prod/job/eos-idp-pr/1/) and pushed the [pr6 tag](https://hub.docker.com/layers/endlessm/eos-idp/pr6/images/sha256-966e6a19173bfe5fb70519fd6ca3a267d30e695d1d889fe4608a4c92c14e307a?context=repo). Merging.
